builds/delete-serverless-service-build-pack/Jenkinsfile

```diff
@@ -44,348 +44,372 @@ import groovy.json.JsonSlurperClassic
-			for (_env in environmentList) {
-				environmentIds.push(_env)
+			if(service_config['provider'] == 'aws'){
+				lambdaEvents.setServiceConfig(service_config)
+			}
+			environmentMetadataLoader.initialize(service_config, configLoader, scmModule, null, env.BUILD_URL, env.BUILD_ID, g_base_url + "/jazz/environments", auth_token)
+			def environmentList = environmentMetadataLoader.getEnvironmentLogicalIds()
+			echo "Environment List: $environmentList"
+
+			if (!events) { error "Can't load events module" } //Fail here
+			events.initialize(configLoader, service_config, "SERVICE_DELETION", "", "", g_base_url + "/jazz/events")
+			context_map = [created_by : service_config['created_by']]
+			sonarModule.initialize(configLoader, service_config, "")
+
+			if (environmentIds.size() > 0) {
+				//validate each id in environment catalog
+				for (_eId in environmentIds) {
+					if (!environmentList.contains(_eId)) {
+						events.sendFailureEvent('INITIALIZE_DELETE_WORKFLOW', "unable to find the environment id $_eId from environment catalog")
+					}
+				}
+			} else if (environmentList.size() > 0) {
+				for (_env in environmentList) {
+					environmentIds.push(_env)
+				}
+			} else if (environmentList.size() == 0) {
+				echo "$repo_name doesn't contain any active environment logical id in environment catalog!"
 			}
-		} else if (environmentList.size() == 0) {
-			echo "$repo_name doesn't contain any active environment logical id in environment catalog!"
-		}
 
-		if (configLoader.TVAULT && configLoader.TVAULT.IS_ENABLED) {
-			vaultUtilityModule.initialize(configLoader, service_config, g_base_url, auth_token, null, events, utilModule)      
-		}
+			if (configLoader.TVAULT && configLoader.TVAULT.IS_ENABLED) {
+				vaultUtilityModule.initialize(configLoader, service_config, g_base_url, auth_token, null, events, utilModule)      
+			}
 
 
-		dir(repo_name){
-			if (fileExists('build.api')) {
-				flowType = "API"
-				loadServerlessConfig()
-				updateSwaggerConfig()
-			} else if (fileExists('build.lambda') || fileExists('build.function')) {
-				flowType = "FUNCTION"
-				loadServerlessConfig()
-			} else if (fileExists('build.website')) {
-				flowType = "WEBSITE"
-				if (service_config['create_cloudfront_url']) {
-					cloudfrontEnabled = service_config['create_cloudfront_url'].trim()
+			dir(repo_name){
+				if (fileExists('build.api')) {
+					flowType = "API"
+					loadServerlessConfig()
+					updateSwaggerConfig()
+				} else if (fileExists('build.lambda') || fileExists('build.function')) {
+					flowType = "FUNCTION"
+					loadServerlessConfig()
+				} else if (fileExists('build.website')) {
+					flowType = "WEBSITE"
+					if (service_config['create_cloudfront_url']) {
+						cloudfrontEnabled = service_config['create_cloudfront_url'].trim()
+					} else {
+						cloudfrontEnabled = "false"
+					}
+				} else if(fileExists('build.slsapp')) {
+					flowType = "SLSAPP"
 				} else {
 					cloudfrontEnabled = "false"
```

Comment:
Instead, initialize this variable during declaration. And, shouldn't this be boolean?

Reply:
Removing the cloudfrontEnabled = "false" from here, since we don't want that line over here

Reply:
And yes it is storing it as string in catalog during service creation.

Reply:
Hmm, I think this flag is not being used below at all (since we will not have websites without CloudFront asset). It is safe to remove this variable.